fix(formatter): missing semicolon for `declare` function

Dunqing opened the pull request:
```ts
// normal function
function a() {}

// declare function
declare function a(): void;
```

`declare function` without a function body, so it behaves like a `declare var a: void`, thus the semicolon is needed

leaysgur commented:
### Merge activity

* **Sep 22, 4:23 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
fix(formatter): missing semicolon for `declare` function (#13928)

```ts
// normal function
function a() {}

// declare function
declare function a(): void;
```

`declare function` without a function body, so it behaves like a `declare var a: void`, thus the semicolon is needed
